pipelines/pipelines.go

```diff
@@ -0,0 +1,196 @@
+package pipelines
+
+import (
+    "context"
+    "sync"
+)
+
+// Chan converts a slice of type T to a channel which receives the provided values. The result channel is closed if the
+// provided Context is cancelled or all values have been sent.
+func Chan[T any](ctx context.Context, in []T) chan T {
```

Comment:
why not buffer the channel, write everything to it, and then close it? would save the goroutine.

Reply:
Nice! Thank you.